### ISSUE TITLE
Allow users choices for starter code

### ISSUE THREAD

**Issue by m-reza-rahman** [eclipse-ee4j/starter#168]
The Starter UI will provide the  user a set of sensible choices for the generated starter code. The initial set of choices will be as following:

```
Please choose your code sample:

(*) Simple RESTful service ❶ ( ) REStful CRUD service ➋ ( ) CRUD web application ❸
```

❶  The default option. Simply generates a RESTful endpoint that prints {“hello”:”world”} in application/json format. Pretty much just Jakarta REST and JSON. No tests included. See here for reference (pretty much already done): https://github.com/m-reza-rahman/starter/tree/master/minimal-starter.

➋  Generates CRUD REST service with just one simple Entity. Shows sample usage of Jakarta REST, JSON, CDI, Persistence, Transactions/Enterprise Beans and Validation. An embedded database is used for convenience (ideally the default one already included in the runtime). A helpful sample test included. See here for reference (pretty much already done): https://github.com/m-reza-rahman/starter/tree/master/rest-starter.

❸  Generates CRUD Faces web application with just one table/screen and one simple entity. No REST endpoint is included. Shows sample usage of Jakarta Faces, CDI, Persistence, Transactions/Enterprise Beans and Validation. An embedded database is used for convenience (ideally the default one already included in the runtime). A helpful sample test included.

**Comment by m-reza-rahman** [comment]
The current consensus is that this is a priority item to be implemented soon. Concise input from key community members can be found here: https://docs.google.com/spreadsheets/d/1bthiYVjX_UK7Xp2qQmq93awPwVhVCDnprGdzeMtv_ow/edit?usp=sharing. A discussion thread on the mailing list can be found here: https://www.eclipse.org/lists/starter-dev/msg00184.html.

**Comment by m-reza-rahman** [comment]
I believe it is important to be deliberate about avoiding too much overlap with the examples project (though clearly there is some overlap): https://github.com/eclipse-ee4j/jakartaee-examples. Both are separate use cases. It's best we focus on the basic getting started experience mostly for beginners and leave less common/more advanced features/options to the examples project instead.

**Comment by OndroMih** [comment]
I think all options should be allowed at any time. For example, Bean Validation could be used even without persistence, e.g. inside a REST resource like here: https://dzone.com/articles/bean-validation-in-jax-rs. I believe that a variant for Faces is also easy to create in a similar way.

**Comment by geziefer** [comment]
> It's best we focus on the basic getting started experience mostly for beginners and leave less common/more advanced features/options to the examples project instead.

I totally agree. That's what the starter is for, to get a qucik start, especially for new or less experienced people. 
It should contain the project structure, the runtime, a test, some basic code examples and a readme with some useful information on how to continue from there. Everything else should be left to a broader example project showing all the relevant Jakarta EE technologies which can then be added to ones project as needed.

**Comment by OndroMih** [comment]
> It's best we focus on the basic getting started experience mostly for beginners and leave less common/more advanced features/options to the examples project instead.

I agree that this is the main focus, to aim at beginners to get started with Jakarta EE quickly. But I hope Starter is there also to help experienced devs to be more productive with Jakarta EE. Therefore in the future, I'd like it to provide examples of more advanced stuff like web.xml, beans.xml, orm.xml (Persistence), as these and some other pieces are not well documented and it's hard to find simple examples. For example, I had to note all the standard persistence.xml options in my blog so that I can easily find them when I need them: https://ondro.inginea.eu/index.php/properties-and-config-in-persistence-xml/. What is also hard to find is XSD schemas for various XMLs, even for persistence.xml, or when you want to create CDI beans.xml from scratch (if your IDE doesn't help you well enough). I like the idea that as an experienced developer, I could go to the Starter and generate an example, from which I can then copy to my project.
